adapters/repos/db/index_queue.go

```diff
@@ -173,6 +173,9 @@ func NewIndexQueue(
 	if opts.BruteForceSearchLimit == 0 {
 		opts.BruteForceSearchLimit = 100_000
 	}
+	if v, err := strconv.Atoi(os.Getenv("ASYNC_BRUTE_FORCE_SEARCH_LIMIT")); err == nil && v >= 0 {
```

Comment:
Won't the `>= 0` mean that we cannot use the env var to effectively turn the brute force off completely using the env var. Because if you set it to `0` it is ignored, if you set it to `1` it is a very small limit, but the code would still run?

Reply:
Which – if what I'm saying is correct – would mean that this

> If set to zero it will disable searching the queued vectors and just return the current HNSW results.

is no longer true.

Reply:
If you set the environment variable to "0", then Atoi returns 0, nil. Since the check is `v >= 0` this will set the bruteForceSearchLimit to 0 see here https://go.dev/play/p/FS9SNxeoDc9.

Reply:
Oh, of course. That makes a lot of sense. Not sure what I thought when I left the comment. Thanks for the example. Approving & merging now.